python/pricing_solver.py

```diff
@@ -0,0 +1,144 @@
+import os
+import pandas as pd
+import logging
+import keys, slope_api
+
+solver_tolerance = 0.001
+solver_max_iterations = 5
+
+solver_folder = "c:\\Slope Api\\Pricing Solver Solver\\"
+
+reports = {
+    "Cash Flows": {"workbook": r'6W5gKlGv2pLW4MwT1UntD8', "element": r'UYyRCGeP3x'},
+    "Pricing": {"workbook": r'5RZWZjMiXjdBj7oPDFdBca', "element": r'd-42wNCNrT'},
+}
+
+class Solver:
+    initial_guess_offset = 0.01                          # Try 1% higher than original projection as first guess
+    target_value_column = 'Pricing Metric Target Value'  # Name of the column in the Entity Parameters table tha has the target IRR
+    commission_value_column = 'Trail Commission Rate'    # Name of the column in the Trail Commission to change in order to solve for target
+
+    def __init__(self, params: {}):
+        self.api = slope_api.SlopeApi()
+        self.api.authorize(keys.api_key, keys.api_secret)
+
+        self.projection_id = params["projection_id"]
+        self.model_id = params["model_id"]
+        self.pricing_target = params["target"]
+
+        self.solver_folder = f"{solver_folder}\\{self.projection_id}\\"
+        if not os.path.exists(self.solver_folder):
+            os.makedirs(self.solver_folder)
+        self.slope_file_path = f"Pricing Solver/{self.projection_id}"
+
+        table_structures = self.api.list_table_structures(self.model_id)
+        self.pricing_table_name = params["pricing_table_name"]
+        pricing_table_structure = next(table for table in table_structures if table["name"] == params["pricing_table_name"])
+        self.pricing_table_structure_id = pricing_table_structure["id"]
+
+        # Set Display preferences for log messages
+        pd.set_option('display.max_columns', None)
+
+    def solve(self):
+        logging.info("Starting Pricing Solver")
+        logging.info(f"  Model ID: {self.model_id}")
+        logging.info(f"  Projection ID: {self.projection_id}")
+        logging.info(f"  Pricing Target: {self.pricing_target}")
+
+        # Get the completed projection details and check status
+        projection_details = self.api.get_projection_details(self.projection_id)
```

Comment:
Presumably this is using the full details because of the bug you previously mentioned? I didn't check the JIRA ticket yet, but please make sure there is scope in the ticket to update this once we fix the API bug, if it's not already in there.